### ISSUE TITLE
shortcut attribute to define document properties

### ISSUE THREAD

**Issue by rdeltour** [xproc/1.0-specification#146]
Can it be considered to add a `properties` attribute to document-loader elements and steps (`p:inline`, `p:document`, `p:load`) to set the document's properties as a shortcut of later invoking `p:set-properties` ?

**Comment by ndw** [comment]
The committee has mixed feelings about doing this now. If it turns out to be very common, then probably. Can you provide some explicit use cases?

**Comment by rdeltour** [comment]
I don't remember the specific use case I had in mind, but basically use cases would be exactly the same where you'd use `p:set-properties`, as it's only a shortcut proposal.
It seemed like a low-hanging fruit, but I understand you need to prioritize.

**Comment by ndw** [comment]
I think the sticking point is that the properties are naturally key/value pairs and having a shortcut attribute to set them would require having to microparse the attribute value. The current standing certainly isn't "no, we're not doing that"; it's more a case of finding out how common the case is whether it's really worth the effort.

**Comment by rdeltour** [comment]
Fair enough. But since properties are `map(xs:string,xs:string)`, don't you have the same micro-parsing issue anyway with the shortcut syntax of the `properties` option on `p:set-properties`?

**Comment by ndw** [comment]
Oh, indeed. Brain cramp on my part. It would just be a map and we have a constructor syntax for maps. #facepalm

**Comment by rdeltour** [comment]
Haha. BTW, I'm not super familiar with XPath 3, but a quick glance shows maps are introduced in XPath **3.1**, and neither 3.0 nor 3.1 are referenced from XProc 2.0. Should #30 be reopened?
